lib/msf/core/exploit/capture.rb

```diff
@@ -163,26 +162,14 @@ def capture_extract_ies(raw)
       end
 
       #
-      # This monstrosity works around a series of bugs in the interrupt
-      # signal handling of Ruby 1.9
+      # Loop through each packet
       #
       def each_packet
         return unless capture
-        begin
-          @capture_count = 0
-          reader         = framework.threads.spawn("PcapReceiver", false) do
-            capture.each do |pkt|
-              yield(pkt)
-              @capture_count += 1
-            end
-          end
-          reader.join
-        rescue ::Exception
-          raise $!
-        ensure
-          reader.kill if reader.alive?
+        capture.each do |pkt|
+          yield(pkt)
+          @capture_count += 1
```

Comment:
@capture_count is not initialized at this point, so we're attempting to increment a nil by one.